resources/test262-agent-harness.js

```diff
@@ -0,0 +1,335 @@
+      throw new Error(e.data[1]);
+    }
+  }));
+  let iframe = document.createElement('iframe');
+  iframe.style = 'display: none';
+  content = test262_as_html(test262, attrs, opts.strict);
+  let blob = new Blob([content], {type: 'text/html'});
+  iframe.src = URL.createObjectURL(blob);
+  document.body.appendChild(iframe);
+
+  let w = iframe.contentWindow;
+  // Finish test on completed event.
+  w.addEventListener('completed', t.step_func(function(e) {
+    t.done();
+  }));
+  // If test failed, rethrow error.
+  let FAILED = 'iframe-failed' + opts.strict ? 'strict' : '';
+  window.addEventListener(FAILED, t.step_func(function(e) {
+    t.set_status(t.FAIL);
+    throw new Error(e.detail);
+  }));
+  // In case of error send it to parent window.
+  w.addEventListener('error', function(e) { 
+    e.preventDefault();
+    // If the test failed due to a SyntaxError but phase was 'early', then the
+    // test should actually pass.
+    if (e.message.startsWith("SyntaxError")) {
+        if (attrs.type == 'SyntaxError' && attrs.phase == 'early') {
+            t.done();
+            return;
+        }
+    }
+    top.dispatchEvent(new CustomEvent(FAILED, {detail: e.message}));
+  });
+}
+
+let HEADER = `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title></title>
+
+  <!-- Test262 harness -->
+  <script src="http://localhost:8000/resources/test262-harness.js"><\/script>
+
+  <!-- Test262 required libraries -->
+  <script src="http://localhost:8000/resources/test262/harness/assert.js"><\/script>
+  <script src="http://localhost:8000/resources/test262/harness/sta.js"><\/script>
+
+  ###INCLUDES###
+
+  <script type="text/javascript">
+    function __completed__() {
+      window.dispatchEvent(new CustomEvent('completed'));
+    }
+  <\/script>
+</head>
+<body>
+</body>
+<script type="text/javascript">
+`;
+let FOOTER = `
+  ;__completed__();
+<\/script>
+</html>
+`;
+
+function test262_as_html(test262, attrs, strict) {
+  output = [];
+  output.push(HEADER.replace('###INCLUDES###', addScripts(attrs.includes)));
+  output.push(prepareTest(test262, attrs, strict));
+  output.push(FOOTER);
+  return output.join("");
+}
+
+function addScripts(sources) {
+  sources = sources || [];
+  let ret = [];
+  let root = 'http://localhost:8000/resources/test262/harness/'
+  sources.forEach(function(src) {
+    ret.push("<script src='###SRC###'><\/script>".replace('###SRC###', root + src));
+  });
+  return ret.join("\n");
+}
+
+function prepareTest(test262, attrs, strict) {
+  function escapeEvalExpr(str) {
+      // Escape double quotes.
+      let ret = str.replace(/\"/g, '\\\"');
+      // Remove backreturn characters.
+      return ret.replace(/\n/g, '');
+  }
+  let output = [];
+  let test = test262();
+  if (strict) {
+    test = "'use strict';\n" + test;
+  }
+  if (attrs.type == 'SyntaxError' && attrs.phase == 'runtime') {
+      // If phase is 'runtime' the error will be caught here. If phase is 'early' the
```

Comment:
`negative:` phases are more specific in the latest version (2.0.0) of https://github.com/tc39/test262/blob/master/INTERPRETING.md, see [negative](https://github.com/tc39/test262/blob/master/INTERPRETING.md#negative) for more details, but the short version is that `early` is now `parse`.

Reply:
You can safely check for both for the time being.